From 021f9306c31b042dd3f507e40ca7b8808e424d93 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@eng.windriver.com>
Date: Thu, 14 Sep 2023 00:50:07 -0700
Subject: [PATCH] fix python3 include directory not found

After python3 in Yocto was upgraded to 3.11, the result of
`importlib.import_module('distutils.sysconfig').get_python_inc()'
is not right, but the result of `importlib.import_module('sysconfig').
get_path('include')' is OK

$ ../recipe-sysroot-native/usr/bin/python3-native/python3
>>> import importlib; import importlib.util;
>>> importlib.import_module('distutils.sysconfig').get_python_inc()
'/tmp-glibc/work/x86_64-linux/tensorflow-native/2.10.0-r0/recipe-sysroot-native/tmp-glibc/work/x86_64-linux/tensorflow-native/2.10.0-r0/recipe-sysroot-native/usr/include/python3.11'
>>> importlib.import_module('sysconfig').get_path('include')
'/tmp-glibc/work/x86_64-linux/tensorflow-native/2.10.0-r0/recipe-sysroot-native/usr/include/python3.11'

Upstream-Status: Inappropriate [Yocto specific]

Signed-off-by: Hongxu Jia <hongxu.jia@eng.windriver.com>
---
 third_party/py/non_hermetic/python_configure.bzl | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/third_party/py/non_hermetic/python_configure.bzl b/third_party/py/non_hermetic/python_configure.bzl
index 300cbfb6c71..14248233dc1 100644
--- a/third_party/py/non_hermetic/python_configure.bzl
+++ b/third_party/py/non_hermetic/python_configure.bzl
@@ -155,11 +155,9 @@ def _get_python_include(repository_ctx, python_bin):
             python_bin,
             "-Wignore",
             "-c",
-            "import importlib; " +
-            "import importlib.util; " +
-            "print(importlib.import_module('distutils.sysconfig').get_python_inc() " +
-            "if importlib.util.find_spec('distutils.sysconfig') " +
-            "else importlib.import_module('sysconfig').get_path('include'))",
+            "from __future__ import print_function;" +
+            "import sysconfig;" +
+            "print(sysconfig.get_path('include'))"
         ],
         error_msg = "Problem getting python include path.",
         error_details = ("Is the Python binary path set up right? " +
-- 
2.40.0

